DeprecatedFile: Eliminate some calls to DeprecatedFile::read_all

BenWiederhake opened the pull request:
This PR removes some, but not all, instances of `DeprecatedFile::read_all`. (One of the remaining instances is #18529.)

The situation in `MemoryStatsWidget::refresh` is really unfortunate, but if the file is corrupted or we OOM while reading a few bytes, then crashing seems like the only reasonable behavior left anyway. Even with some mechanism to propagate these errors across the EventLoop, I don't see how SystemMonitor could possibly react any better.

Similarly for `MulticastDNS::local_addresses()`: If the Kernel responds with incomplete or broken JSON, all hope is lost. But for LookupServer there is at least some point in trying to stay alive as other processes might depend on it still running, hence my overkil.

Advances #17129.

LucasChollet commented on Userland/Libraries/LibCodeComprehension/Cpp/Tests.cpp:
```diff
@@ -89,9 +89,9 @@ int run_tests()
-    VERIFY(!file.is_error());
-    filedb.add(name, DeprecatedString::copy(file.value()->read_all()));
+    auto file_or_error = Core::File::open(LexicalPath::join(TESTS_ROOT_DIR, name).string(), Core::File::OpenMode::Read);
+    VERIFY(!file_or_error.is_error());
```
Isn't that the same as `realese_value_but_fixme_should_propagate_errors`?

BenWiederhake replied:
It's inside a test, so there's no real destination where to report an error. So it isn't really a FIXME, because it would still abort the test. I don't really like the idea of adding an unfixable fixme. We could make it a "MUST", what do you think about that?

And just in advance: `TRY_OR_FAIL` doesn't work here because this test suite wasn't written in the "return ErrorOr" style. Introducing it now would require much more work, and doesn't fit into a PR about DeprecatedFile migration.

BenWiederhake commented:
Changes since last push:
- In the LibCodeComprehension test, we now replace the `VERIFY(!.iserror())` with a `MUST`. (As explained, `TRY_OR_FAIL` requires too much off-topic work for this PR.)
- `Core::command(…)` now returns ByteBuffers, which actually simplifies this PR.

Note that the Azure CI jobs fail because all of them fail: https://github.com/SerenityOS/serenity/issues/18887

GMTA commented on Userland/Libraries/LibCodeComprehension/Cpp/Tests.cpp:
```diff
@@ -89,9 +89,8 @@ int run_tests()
-    VERIFY(!file.is_error());
-    filedb.add(name, DeprecatedString::copy(file.value()->read_all()));
+    auto file = MUST(Core::File::open(LexicalPath::join(TESTS_ROOT_DIR, name).string(), Core::File::OpenMode::Read));
+    filedb.add(name, DeprecatedString::copy(MUST(file->read_until_eof())));
```
You should use `MUST` if there should be no chance that an operation fails.

Here, it is more appropriate that you use `.release_value_but_fixme_should_propagate_errors()`.

BenWiederhake replied:
As I already responded to Lucas, this is not a fixme, as it is unfixable: The only correct behavior is to abort the test.

But okay, I guess I'll add a fixme then :shrug:

GMTA replied:
I missed both that discussion _and_ the fact that this was part of test cases. Why is this not in `Tests/`? :cry:

Anyway, using `MUST` here makes sense in the context of tests. Although the commit message should be updated to reflect either the usage of `MUST` or `release_value_but_fixme_should_propagate_errors` if you choose to leave it that way.

BenWiederhake replied:
How about we leave it as a release_value_but_fixme_should_propagate_errors, and in a *different* PR I'll move the file to the Tests folder and change it to a MUST?

GMTA replied:
Sounds great! As long as the commit message reflects the changes in this PR though :^)

BenWiederhake replied:
:+1: There already was *some* description, I expanded upon it a bit.

BenWiederhake commented:
Changes since last push:
- Rebased for convenience
- Changed a "MUST" to a "release_value_but_fixme_should_propagate_errors". (Even though there is no other "more correct" behavior, so this is not really a fixme, because it cannot be "fixed". However, I don't care enough to continue that discussion, so I'll go with "release_value_but_fixme_should_propagate_errors".)

BenWiederhake commented:
Changes since last push:
- Rebased for convenience
- Documented the choice of "release_value_but_fixme_should_propagate_errors" in the commit message a bit more.
